### DISCUSSION TITLE
Restoring teslamate backup from existing install - no back file

### DISCUSSION THREAD

**Discussion by fergymcferguson** [teslamate-org/teslamate#3473] [Q&A]
Hello,  I have a bit of a problem.  some recent power outages have made the SD card in my raspberry pi un-bootable.  I technically do have a teslamate.bck file but it's outdated by 1.5 years so I would lose that much data collection from the car :(  I thought I had a crontab setup to run a backup weekly, but unfortunately I never double checked that it was working properly and the backup ran exactly once. 

I am able to mount the SD card and access the directories including /home/pi and /var/lib/docker

I have created a new install of raspian and my plan was to move over my docker-compose.yml file and the entire /var/lib/docker directory, then start the docker containers back up again.  Will this work, or am I kind of out of luck?

Thanks in advance.

**Comment by eliluong** [comment]
You can try asking around Linux or RPi forums to see if you can fix it (Rpi) to make it bootable again.

The Docker compose file creates a named volume for postgres database, `teslamate-db:/var/lib/postgresql/data`, which in this case is `teslamate-db`. I forget where this gets stored when you have a named volume (versus pointing it at a specific directory, a bind mount). But if you are able to find it (some Google, chatgpt, etc), copy it to the same directory on another computer, then run just a new postgres database container in Docker and point to the same directory for the data, you should be able to run the backup command. Then you can run the full stack of teslamate/postgress/mqtt/etc and restore.

Also if you keep all the image versions the same (postgres, etc), you might be able to just copy/paste like you wanted instead of going through backup/restore.
